props/websubmit/src/main.rs

```diff
@@ -85,16 +93,98 @@ impl DeletionProp {
     }
 }
 
+/// Storing data in the database must be associated to a user. This is
+/// necessary for e.g. the deletion to work.
+pub struct ScopedStorageProp {
+    cx: Arc<PolicyContext>,
+}
+
+impl ScopedStorageProp {
+    pub fn new(cx: Arc<PolicyContext>) -> Self {
+        ScopedStorageProp { cx }
+    }
+
+    pub fn check(self) {
+        for c_id in self.cx.desc().controllers.keys() {
+            let scopes = self
+                .cx
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|node| self.cx.has_marker(marker!(scopes), *node))
+                .collect::<Vec<_>>();
+            let stores = self
+                .cx
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|node| self.cx.has_marker(marker!(stores), *node))
+                .collect::<Vec<_>>();
+            let mut sensitives = self
+                .cx
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|node| self.cx.has_marker(marker!(sensitive), *node));
+
+            let controller_valid = sensitives.all(|sens| {
+                stores.iter().all(|&store| {
+                    // sensitive flows to store implies some scope flows to store callsite
+                    !(self
+                        .cx
+                        .flows_to(sens, store, paralegal_policy::EdgeType::Data))
+                        || store.typ.as_call_site().is_some_and(|cs| {
+                            scopes.iter().any(|scope| {
+                                self.cx.flows_to(
+                                    *scope,
+                                    Node {
+                                        ctrl_id: store.ctrl_id,
+                                        typ: cs.into(),
+                                    },
+                                    paralegal_policy::EdgeType::Data,
+                                )
+                            })
+                        })
+                })
+            });
+
+            assert!(controller_valid);
+        }
+    }
+}
+
 fn main() -> Result<()> {
     let ws_dir = std::env::args()
         .nth(1)
-        .ok_or_else(|| anyhow!("expected an argument"))?;
-    paralegal_policy::SPDGGenCommand::global()
-        .run(ws_dir)?
-        .with_context(|ctx| {
+        .ok_or_else(|| anyhow!("expected format: cargo run <path> [policy]"))?;
+    let prop_name = match std::env::args().nth(2) {
```

Comment:
I would not use "" here, instead return `Option<String>`. Then run the combined on `None` and the others on `Some(name)` respectively but if the `name` is not `"sc"` or `"del"` I would error.
Alternatively you can also make an `enum` for those and use the `strum` library to derive `EnumString` that parses them from their names.
If you want to be really fancy you could also use `clap` for the arguments (it works together with the `FromStr` that `strum` generates)

Reply:
Done